File: string-paradigm-expression-original/src/main/java/com/jesperancinha/string/paradigm/original/expression/SpellingCheck.java
package com.jesperancinha.string.paradigm.original.expression;

import java.util.ArrayList;
import java.util.List;
import java.util.regex.Pattern;

public class SpellingCheck {

    private static final Pattern DOUBLE_COMMAS = Pattern.compile(".*,\\s*,.*");
    private static final Pattern WHITE_SPACES = Pattern.compile(".*\\s\\s+.*");

    private final List<String> spellingDoubleCommaRegions = new ArrayList<String>();
	private final List<String> spellingWhiteSpacesRegions = new ArrayList<String>();
	private final List<String> spellingBeginCapitalRegions = new ArrayList<String>();

    public void checkSpelling(String text) {
    	if (DOUBLE_COMMAS.matcher(text).matches()) {
    	    spellingDoubleCommaRegions.add(text);
    	}
    	if (WHITE_SPACES.matcher(text).matches()) {
    	    spellingWhiteSpacesRegions.add(text);
    	}
    	String[] elements = text.split(",");
    	for (int i = 0; i < elements.length; i++) {
    		String element = elements[i].trim();
    		if (element.length() > 0 && Character.getType(element.charAt(0)) == Character.LOWERCASE_LETTER) {
    		    spellingBeginCapitalRegions.add(text);
    		}
    	}
    }

	public List<String> getSpellingDoubleCommaRegions() {
		return spellingDoubleCommaRegions;
	}

	public List<String> getSpellingWhiteSpacesRegions() {
		return spellingWhiteSpacesRegions;
	}

	public List<String> getSpellingBeginCapitalRegions() {
		return spellingBeginCapitalRegions;
	}
}

File: string-paradigm-expression-original/src/main/kotlin/org/jesperancinha/string/paradigm/original/expression/SpellingCheck.kt
package org.jesperancinha.string.paradigm.original.expression

import java.util.regex.Pattern

class SpellingCheck {
    val spellingDoubleCommaRegions: MutableList<String> = ArrayList()
    val spellingWhiteSpacesRegions: MutableList<String> = ArrayList()
    val spellingBeginCapitalRegions: MutableList<String> = ArrayList()
    fun checkSpelling(text: String) {
        if (DOUBLE_COMMAS.matcher(text).matches()) {
            spellingDoubleCommaRegions.add(text)
        }
        if (WHITE_SPACES.matcher(text).matches()) {
            spellingWhiteSpacesRegions.add(text)
        }
        val elements = text.split(",".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
        for (i in elements.indices) {
            val element = elements[i].trim { it <= ' ' }
            if (element.length > 0 && Character.getType(element[0]) == Character.LOWERCASE_LETTER.toInt()) {
                spellingBeginCapitalRegions.add(text)
            }
        }
    }

    companion object {
        private val DOUBLE_COMMAS = Pattern.compile(".*,\\s*,.*")
        private val WHITE_SPACES = Pattern.compile(".*\\s\\s+.*")
    }
}